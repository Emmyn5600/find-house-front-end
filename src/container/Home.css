.hRZzmY {
  display: grid;
  grid-column: 1;
  background: #c4bdac;
  box-shadow: 0 4px 6px rgb(0 0 0 / 7%);
  border-radius: 12px;
  border: 0 solid transparent;
  margin: 1rem;
  height: auto;
  align-items: flex-start;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0;
  flex-wrap: wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
}

.hRZzmY .border-line {
  height: 7px;
  width: 100%;
  background: linear-gradient(89.35deg, #ffa05c 0%, #ff7958 99.99%);
  overflow: hidden;
  border-top-left-radius: 12px;
  border-top-right-radius: 12px;
}

.hRZzmY.pm .border-line {
  background: #1c252e;
}

.hRZzmY .blocks-container {
  padding: 35px;
  overflow: hidden;
}

.hRZzmY:hover {
  box-shadow: 0 15px 30px #4c2818;
}

.hRZzmY img {
  height: 230px;
  width: 400px;
}

.hRZzmY img:hover {
  -ms-transform: scale(1.5); /* IE 9 */
  -webkit-transform: scale(1.5); /* Safari 3-8 */
  transform: scale(1.2);
  transition: width 400ms, height 400ms, background-color 400ms, transform 400ms;
}

.hRZzmY .title {
  font-size: 20px !important;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", "Liberation Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
  font-weight: 600;
  color: #212b36;
  font-style: normal;
  font-stretch: normal;
  line-height: 2;
  text-align: left;
  margin-top: 20px !important;
}

.hRZzmY .description {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", "Liberation Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
  font-size: 14px;
  font-weight: 600;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.4;
  min-height: 117px;
  -webkit-letter-spacing: 0.2px !important;
  -moz-letter-spacing: 0.2px !important;
  -ms-letter-spacing: 0.2px !important;
  letter-spacing: 0.2px !important;
  text-align: left;
  color: #28292e;
  text-decoration: none;
}

.price {
  color: #000;
  background-color: #5e544e;
}

.hRZzmY.os .border-line {
  background: #1c252e;
}

.hRZzmY.api .border-line {
  background: #1c252e;
}

.hRZzmY.ui .border-line {
  background: #1c252e;
}

.hRZzmY.cli .border-line {
  background: #1c252e;
}

.hRZzmY.ds .border-line {
  background: #1c252e;
}

.button {
  font-weight: 700;
  color: #000;
  text-align: center;
  border-radius: 6px;
  width: 170px;
  height: 40px;
  font-size: 15px;
  text-transform: uppercase;
  cursor: pointer;
  margin-bottom: 20px;
  background-color: #daf6e6;
}

.button:hover {
  color: #a17362;
}
