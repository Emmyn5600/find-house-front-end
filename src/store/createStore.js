import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import reducer from './reducers/index';

const store = () => createStore(reducer, applyMiddleware(thunk));

export default store;
