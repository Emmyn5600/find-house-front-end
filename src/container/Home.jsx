import React, { useState, useRef, useEffect } from 'react';
import { ThemeProvider } from 'styled-components';
import { connect } from 'react-redux';
import jwtDecode from 'jwt-decode';
import PropTypes from 'prop-types';
import { useOnClickOutside } from '../hooks';
import { GlobalStyles } from '../helper/global';
import { theme } from '../helper/theme';
import { Burger, Menu } from '../components';
import { signInUserSuccess } from '../store/actions/actionCreators';
import { loadHousesAsync } from '../store/thunk-redux/housesThunk';
import House from '../components/House';
import './Home.css';

function Home({
  history, signinUser, loadHouses, houses,
}) {
  const [open, setOpen] = useState(false);
  const node = useRef();
  useEffect(() => {
    const token = localStorage.getItem('authToken');
    if (!token) history.replace('/login');
    if (token) {
      const user = jwtDecode(JSON.parse(token));
      signinUser(user);
    }
    loadHouses();
  }, []);
  useOnClickOutside(node, () => setOpen(false));
  const handleToggleMenu = (state) => {
    setOpen(state);
  };
  return (
    <section>
      <ThemeProvider theme={theme}>
        <>
          <GlobalStyles />
          <Burger setOpen={handleToggleMenu} open={open} />
          <Menu open={open} />
          {houses.map((house) => {
            /* eslint arrow-body-style: */
            return (
              <House house={house} key={house.id} />
            );
          })}
        </>
      </ThemeProvider>
    </section>
  );
}

Home.propTypes = {
  signinUser: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.any).isRequired,
  loadHouses: PropTypes.func.isRequired,
  houses: PropTypes.arrayOf(PropTypes.any).isRequired,
};

const mapStateToProps = (state) => ({
  loading: state.houses.loading,
  houses: state.houses.list,
});

const mapDispatchToProps = {
  signinUser: (user) => signInUserSuccess(user),
  loadHouses: () => loadHousesAsync(),
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
