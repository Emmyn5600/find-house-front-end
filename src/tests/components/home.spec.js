import { BrowserRouter, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import createStore from '../../store/createStore';
import Home from '../../container/Home';

const store = createStore();

describe('Home', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Route path="/" component={Home} />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should not render an unknown link', () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Route path="/" component={Home} />
        </Provider>
      </BrowserRouter>,
    );
    expect(() => screen.getByText(/cool/i)).toThrow();
  });

  it('should throw an error if looking for an unexisting element', () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Route path="/" component={Home} />
        </Provider>
      </BrowserRouter>,
    );
    expect(() => screen.getByText(/soft/i)).toThrow();
  });

  it('should not render an unknown link', () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Route path="/" component={Home} />
        </Provider>
      </BrowserRouter>,
    );
    expect(() => screen.getByText(/cool/i)).toThrow();
  });
});
