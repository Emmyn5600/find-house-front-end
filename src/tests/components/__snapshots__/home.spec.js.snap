// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
Array [
  <button
    className="sc-bdfBQB dQQSnr"
    onClick={[Function]}
    open={false}
  >
    <div />
    <div />
    <div />
  </button>,
  <nav
    className="sc-gsTEea fpQRFU"
    open={false}
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/rents"
      onClick={[Function]}
    >
      My Rents
    </a>
    <a
      href="/logout"
      onClick={[Function]}
    >
      Logout
    </a>
  </nav>,
]
`;
